src/counter/updating.md

```diff
@@ -96,3 +96,33 @@ near dev-deploy $(raen build --release -q)
 You'll see that this takes longer to build than when you make changes to your contract itself. Rust needs to recompile all of your dependencies using the new setting.
 
 Refresh your admin panel, set the number back to `127`, and call `increment`. You'll see an error message. This means the contract panicked rather than allowing the integer overflow. It worked!
+
+## Digging Deeper: wait, aren't blockchains immutable?
+
+You may have heard that blockchains are _immutable_, that they provide permanent ledgers of all on-chain activity.
+
+How does this fit with the fact that you just upgraded a smart contract, and it was actually super easy? It's even called a _contract_—doesn't that imply that it should be hard to change? If it's so easy for a contract owner to alter the contract, how can anyone using it trust they won't be scammed?
+
+Great questions!
+
+On other blockchains, it is in fact very difficult to upgrade a contract. They work around this with proxy contracts and other patterns.
+
+But NEAR has that handy account system, where each account has multiple keys. We already learned about _Function Call Access Keys_ two units ago; that's what makes it safe to have a web-based wallet, with web apps that store a private key in local storage.
```

Comment:
```suggestion
But NEAR's handy account system, where each account can have multiple keys. We already learned about _Function Call Access Keys_ two units ago; that's what makes it safe to have a web-based wallet, with web apps that store a restricted private key in a browser's local storage.
```

Reply:
Thanks; I incorporated this changes slightly differently